### ISSUE TITLE
Riley Link communication fails. Restart Riley needed

### ISSUE THREAD

**Issue by ghost** [LoopKit/Loop#1166]
Running Riley Link for almost a year. It is very helpful managing diabetes. But since updating my iPhone 8 to iOS 13 (currently 13.2) I run into a problem with Riley Link. Randomly the connection between Riley Link and Loop stops working. The only way to get it back working is by powering off Riley Link with the switch on the Riley Link and then startup again. Any action on the iPhone like turning BT on and off, or use the toggle switch in Loop to stop communication is not working. Any idea what could be the problem?

Best regards,

Frans

**Comment by thebrok2** [comment]
I recently have been having/had this issue or a similar issue as well. Suddenly I had to restart my rileylink a couple of times a day. Have tried disabling Bluetooth, restarting the phone, relaunching Dexcom and loop with no luck other than restarting rileylink. Pretty new rileylink (about 3 months) 

--RileyLink Info --
Firmware:  subg_rfspy 2.2 / ble_rfspy2.0
signal strength: -81db
frequency 433.9097900390625 MHz

--Phone--
Iphone XR


When trying to play test beep I would get the following error message. 
peripheralmanagererror(rileylinkblekit.prepheralmanagererror.timeout)

(screenshot: https://share.icloud.com/photos/011QQFP9T-V1H5Iip1v_7M3Rw)

Noticed it right after changing pods (using omnipod). I was not sure if it was the newer pod, a Bluetooth issue, or something else. Appeared to be a Bluetooth issue. 

I just applied the most recent IOS update 13.2 Was running 13.1. I have not had the issue since updating  to 13.2 but it has been less than 24 hours so time will tell. Don't see any notes about Bluetooth fixes in 13.2 however, I have heard some talk about Dexcom Bluetooth issues being fixed with the 13.2 update. 

Not sure if it is a fluke but noticed similar symptoms recently.

**Comment by FransLuijendijk** [comment]
For what it is worth. My RileyLink is running on the same firmware. Dropped the question at the Riley website as well. It feels like iOS is messing up with the RileyLink firmware causing a drop of BT connection although everything looks okay and RileyLink still seems connected.

**Comment by FransLuijendijk** [comment]
Updating Loop to the latest version dev Branch v1.10.1 unfortunately didn't solve the problem (but it does have some nice features). Riley is on Firmware: subg_rfspy 2.2 / ble_rfspy2.0 with a signal strength around -81dB. Not sure what else can be done.

**Comment by loyalchallenger** [comment]
I encounter the same issue quite often. Most recently, after my pod change last night. I lost connection and had to toggle the Rileylink on/off to regain connection.

**Comment by dm61** [comment]
@FransLuijendijk next time you encounter the issue, but _before restarting_ RileyLink, please capture the Loop RileyLink screen (the screen with RL name, Firmware, Signal Strength, Connection State, Uptime etc.), as well as the Loop Issue Report (which is near the top of the Settings screen), then post the screenshot and the report here. Someone may then be able to look into where the comm problems are coming from.

**Comment by FransLuijendijk** [comment]
@dm61 Will do Dragan.

**Comment by FransLuijendijk** [comment]
Over the last weeks I had the issue only twice after the update to the latest version of Loop. The Loop Issue Report is attached 

[Loop Report 2019-11-19 06:57:34+01:00.pdf](https://github.com/LoopKit/Loop/files/3869250/Loop.Report.2019-11-19.06.57.34%2B01.00.pdf)

**Comment by jfullerton3** [comment]
I have been having the same issue since updating to to the latest master branch on loop. This started when I was on IOS12 and has gotten worse when I upgraded to iOS13.

**Comment by human911** [comment]
I'm in the same boat. I was pulling 20+ day of up-time with my RileyLink. 
Now after the iOS 13 update... I will have to physically power on and off the RileyLink every 12-24 hours.

iPhone XR - iOS 13.3
Loop: 1.9.6
RL: subg_rfspy 2.2 / ble_rfspy 2.0

I receive the "Peripheral did not respond in time" when I click on the red loop.
Will also try to capture the screenshot @dm61 is requesting above and attached my logs here.

**Comment by danno-loop** [comment]
Same issue here.

dev Branch v1.10.1
Riley Firmware: subg_rfspy 2.2 / ble_rfspy2.0
OmniPod
Iphone 6S - IOS 13.1.3

I also notice that I get periodic drops throughout day and then re-close of loop after 10-15 minutes without having to reset anything, almost as if losing signal even though green light is on and db levels seem to be below 80, all with RL in my pocket and Iphone with in a few feet of me

**Comment by live4sw** [comment]
Same config as danno-loop except I am running an iPhone SE with iOS 13.3. For the past week or so, my RL has stopped working every couple of days and needs to be turned off and on to work again. Green light still on, db levels good too, RL in pocket and iPhone close by. I was kind of chalking it up to winter and static electricity but maybe there is an actual issue. I will try to capture an issue report also next time it happens.

**Comment by dm61** [comment]
@human911, @live4sw, @danno-loop RL freezing more often after upgrade to IOS 13 has been discussed on Zulip here: https://loop.zulipchat.com/#narrow/stream/148542-RileyLink/topic/Rileylink.20freezing. A distinguishing symptom is that the RileyLink page in Loop shows only one firmware (ble_rfspy) instead of two (subg_rfspy and ble_rfspy) - see the screenshot captured by Katie in her first RL-freezing post on Zulip.  Although there is no known solution to this particular problem, it seems to subside after some time, so you may just need to keep iOS up to date and endure through a period of having to power RL off and on more often. After I made my post on Zulip on Dec 7 I've had only one more case of RL freezing, everything seems to be back in good working order.

**Comment by human911** [comment]
See attached Screenshots and Issue report.
@dm16 Thanks for the info I will look get a screenshot of the rileylink page in Loop next time.


![image0](https://user-images.githubusercontent.com/10080856/71501105-3cf3e180-281d-11ea-82f7-62a31f89cb4a.png)
[Loop Report 2019-12-26 17_28_09-08_00.zip](https://github.com/LoopKit/Loop/files/4004176/Loop.Report.2019-12-26.17_28_09-08_00.zip)

**Comment by dm61** [comment]
@human911 yes, that's the error you'd typically see when RL is in the frozen state so that it has to be powered off/on to resume normal operation.

**Comment by human911** [comment]
@dm61 iOS 13.3.1 dropped lets see if that helps the comm issues,

**Comment by robinwindsorharker** [comment]
This is a pretty common occurrence that is usually solved by turning the rl off and on, but we just upgraded to auto bolus and I’ve restarted the rl 5 times, the phone, the app, and I’m still getting the peripheral timeout error.

**Comment by FransLuijendijk** [comment]
Today connection dropped between RL and the App but I simply let it go. 4 hours later I had to change my Dexcom sensor. 20 minutes after starting the new sensor I got an alert from Loop stating that it could not 'Loop' for 20 minutes. As soon as G6 was ready, so was Loop. Back online without a restart from RL. Strange isn't it. It looks like Dexcom communication with RL might have to do with the problem. Anyone else seen this happen?
